### PR TITLE
Fix(Agent): fix CSS and readonly input

### PR DESCRIPTION
**Pull request by stonebuzz**
Abandon ```generic_show_form.html.twig``` to allow read only on native fields such as ```name``` or ```agenttypes_id```.

Before : 

image

After : 

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: templates/pages/admin/inventory/agent.html.twig [hunk 1]
```diff
@@ -31,145 +31,214 @@
...
+                        'name',
+                        item.fields['name'],
+                        __('Name'),
+                        field_options
+                    ) }}
+
+                    {{ fields.htmlField(
+                        'agenttypes_id',
+                        call('Dropdown::getDropdownName', ["glpi_agenttypes", item.fields['agenttypes_id']]),
+                        __('Agent type'),
```
Comment (trasher) [inline templates/pages/admin/inventory/agent.html.twig:72]:
```suggestion
                        _n('Agent type', 'Agents types', 1),
```

**Review by trasher — APPROVED** [review]

File: templates/pages/admin/inventory/agent.html.twig [hunk 2]
```diff
@@ -31,145 +31,214 @@
...
+                    {{ fields.htmlField(
+                        "Agent",
+                        item.fields['remote_addr'],
+                        __('Public contact address'),
+                        field_options
+                    ) }}
+
+                    {{ fields.htmlField(
+                        "itemtype",
+                        call(item.fields['itemtype'] ~ '::getTypeName', [0]),
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:106]: There is a Twig filter for this.
```suggestion
                        item.fields['itemtype']|itemtype_name,
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 3]
```diff
@@ -31,145 +31,214 @@
...
+                                        general
+                                    ]
+                                })
+                            ) }}
+                        {% endfor %}
+
+                        <div class="hr-text">
+                            <i class="ti ti-list"></i>
+                            <span>{{ __('Supported tasks') }} </span>
+                            <span class="form-help" data-bs-toggle="popover" data-bs-placement="top" data-bs-html="true" data-bs-content="<i> {{ __('When reported by GLPI agent via native inventory') }}</i>">?</span>
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:193]: I think that HTML tags should be escaped inside the attribute value.
```suggestion
                            <span class="form-help" data-bs-toggle="popover" data-bs-placement="top" data-bs-html="true" data-bs-content="{{ '<i>' ~ __('When reported by GLPI agent via native inventory') ~ '</i>' }}">?</span>
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 4]
```diff
@@ -31,145 +31,214 @@
...
+                    {{ fields.htmlField(
+                        'name',
+                        item.fields['name'],
+                        __('Name'),
+                        field_options
+                    ) }}
+
+                    {{ fields.htmlField(
+                        'agenttypes_id',
+                        call('Dropdown::getDropdownName', ["glpi_agenttypes", item.fields['agenttypes_id']]),
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:71]: There is a Twig function for this.
```suggestion
                        get_item_name('AgentType', item.fields['agenttypes_id']),
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 5]
```diff
@@ -31,145 +31,214 @@
...
+            </div> {# .row #}
+         </div> {# .flex-row #}
+      </div>
+      {% if item_has_pictures %}
+         <div class="col-12 col-xxl-3 flex-column">
+            <div class="flex-row asset-pictures">
+               {{ include('components/form/pictures.html.twig', {'gallery_type': ''}) }}
+            </div>
+         </div>
+      {% endif %}
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:226]:
```suggestion
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 6]
```diff
@@ -31,145 +31,214 @@
...
+
+<div class="asset {{ bg }}">
+   {{ include('components/form/header.html.twig', {'in_twig': true}) }}
+
+   {% set rand = random() %}
+   {% set params  = params ?? [] %}
+   {% set target       = params['target'] ?? item.getFormURL() %}
+   {% set withtemplate = params['withtemplate'] ?? '' %}
+   {% set item_type = item.getType() %}
+   {% set item_has_pictures = item.hasItemtypeOrModelPictures() %}
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:50]: The code related to pictures could be cleaned. Agent has no pictures.
```suggestion
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 7]
```diff
@@ -31,145 +31,214 @@
...
+   {% set field_options = {
+       'locked_fields': item.getLockedFields(),
+       'rand': rand,
+   } %}
+
+   <div class="card-body d-flex flex-wrap">
+      <div class="col-12 col-xxl-{{ item_has_pictures ? '9' : '12' }} flex-column">
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:57]:
```suggestion
      <div class="col-12 flex-column">
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 8]
```diff
@@ -31,145 +31,214 @@
...
+            <div class="flex-row asset-pictures">
+               {{ include('components/form/pictures.html.twig', {'gallery_type': ''}) }}
+            </div>
+         </div>
+      {% endif %}
+   </div> {# .card-body #}
+
+   {% if item_type == 'Contract' %}
+      {{ include('components/form/support_hours.html.twig') }}
+   {% endif %}
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:231]:
```suggestion
```

File: templates/pages/admin/inventory/agent.html.twig [hunk 9]
```diff
@@ -31,145 +31,214 @@
...
+   {% if no_form_buttons is not defined or no_form_buttons == false %}
+      {{ include('components/form/buttons.html.twig') }}
+   {% endif %}
+   {% if no_inventory_footer is not defined or no_inventory_footer == false %}
+      {{ include('components/form/inventory_info.html.twig') }}
+   {% endif %}
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:237]:
```suggestion
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**